pyvista/plotting/composite_mapper.py

```diff
@@ -826,7 +826,7 @@ def set_scalars(
 
         self.scalar_visibility = True
         if rgb:
-            self.scalar_mode = 'direct'
+            self.color_mode = 'direct'
```

Comment:
It's technically seperate from the MRO change but it was causing a number of test failures against the latest version of VTK since this was incorrect and now `scalar_mode` exists upstream. Same for the other change

Reply:
OK, thanks. I'm just surprised because I can't see `scalar_mode` defined anywhere now, and it doesn't seem to have been removed in this diff, so I'm surprised how it worked earlier :smile:

Reply:
I'm pretty sure it didn't work 😉

Reply:
These composite rendering features seem to be sparsely used

Reply:
This change should probably be further investigated. There was a bug with composite mappers reported in #6012, resolved by #6272. But, that bug and fix should not be necessary based on the VTK docs, see this discussion https://github.com/pyvista/pyvista/pull/6272#discussion_r1646670531.

I wonder if this method (i.e. `set_scalars`) was intended to be called by the plotter somewhere but isn't. Perhaps it's possible that calling this method would fix #6012, and #6272 could be reverted.